platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/LinkedEventIntake.java

```diff
@@ -0,0 +1,225 @@
+    }
+
+    /**
+     * Add an event to the hashgraph
+     *
+     * @param event an event to be added
+     */
+    public void addEvent(final EventImpl event) {
+        try {
+            // an expired event will cause ShadowGraph to throw an exception, so we just to discard it
+            if (consensusSupplier.get().isExpired(event)) {
+                return;
+            }
+
+            dispatcher.preConsensusEvent(event);
+
+            logger.debug(INTAKE_EVENT.getMarker(), "Adding {} ", event::toShortString);
+            final long minGenNonAncientBeforeAdding = consensusSupplier.get().getMinGenerationNonAncient();
+
+            // Prehandle transactions on the thread pool.
+            prehandlePool.submit(buildPrehandleTask(event));
+
+            // record the event in the hashgraph, which results in the events in consEvent reaching consensus
+            final List<ConsensusRound> consensusRounds = consensusSupplier.get().addEvent(event);
+
+            dispatcher.eventAdded(event);
+
+            if (consensusRounds != null) {
+                consensusRounds.forEach(this::handleConsensus);
+            }
+
+            if (consensusSupplier.get().getMinGenerationNonAncient() > minGenNonAncientBeforeAdding) {
+                // consensus rounds can be null and the minNonAncient might change, this is probably because of a round
+                // with no consensus events, so we check the diff in generations to look for stale events
+                handleStale(minGenNonAncientBeforeAdding);
+            }
+        } finally {
+            intakeEventCounter.eventExitedIntakePipeline(event.getBaseEvent().getSenderId());
+        }
+    }
+
+    /**
+     * Build a task that will prehandle transactions in an event. Executed on a thread pool.
+     *
+     * @param event the event to prehandle
+     */
+    @NonNull
+    private Runnable buildPrehandleTask(@NonNull final EventImpl event) {
+        return () -> {
+            prehandleEvent.accept(event);
+            event.signalPrehandleCompletion();
+        };
+    }
+
+    /**
+     * Notify observer of stale events, of all event in the consensus stale event queue.
+     *
+     * @param previousGenerationNonAncient the previous minimum generation of non-ancient events
+     */
+    private void handleStale(final long previousGenerationNonAncient) {
+        // find all events that just became ancient and did not reach consensus, these events will be considered stale
+        final Collection<EventImpl> staleEvents = shadowGraph.findByGeneration(
+                previousGenerationNonAncient,
+                consensusSupplier.get().getMinGenerationNonAncient(),
+                LinkedEventIntake::isNotConsensus);
+
+        for (final EventImpl staleEvent : staleEvents) {
+            staleEvent.setStale(true);
+            dispatcher.staleEvent(staleEvent);
+            logger.warn(STALE_EVENTS.getMarker(), "Stale event {}", staleEvent::toShortString);
```

Comment:
This is either test only logging (most likely the case) or a spam risk.

Reply:
Would you prefer removal, or rate limited logging?

Reply:
My preference is removal.